find-cycles.hs

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i runghc -p "ghc.withPackages (ps: [ ps.hiedb ps.temporary ])" -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/8cad3dbe48029cb9def5cdb2409a6c80d3acfe2e.tar.gz
```

Comment:
Can this new script live in `src/find-cycles/find-cycles.hs` (analogous to `agda-bisect`)?
Then one could also add a `.cabal` file there to have it run by non-nixers.

Reply:
Done in 02baab9